packages/fiber/src/core/renderer.ts

```diff
@@ -314,68 +328,55 @@ function createRenderer<TCanvas>(roots: Map<TCanvas, Root>, getEventPriority?: (
         return null
       }
     },
-    commitUpdate(
-      instance: Instance,
-      [reconstruct, diff]: [boolean, DiffSet],
-      type: string,
-      oldProps: InstanceProps,
-      newProps: InstanceProps,
-      fiber: Reconciler.Fiber,
-    ) {
+    commitUpdate(instance, [reconstruct, diff]: [boolean, DiffSet], type, _oldProps, newProps, fiber) {
       // Reconstruct when args or <primitive object={...} have changes
       if (reconstruct) switchInstance(instance, type, newProps, fiber)
       // Otherwise just overwrite props
       else applyProps(instance, diff)
     },
-    commitMount(instance: Instance, type, props, int) {
+    commitMount(instance, _type, _props, _int) {
       // https://github.com/facebook/react/issues/20271
       // This will make sure events are only added once to the central container
       const localState = (instance.__r3f ?? {}) as LocalState
       if (instance.raycast && localState.handlers && localState.eventCount) {
         instance.__r3f.root.getState().internal.interaction.push(instance as unknown as THREE.Object3D)
       }
     },
-    getPublicInstance: (instance: Instance) => instance,
-    shouldDeprioritizeSubtree: () => false,
+    getPublicInstance: (instance) => instance,
     prepareForCommit: () => null,
-    preparePortalMount: (container: UseBoundStore<RootState>) => prepare(container.getState().scene),
+    preparePortalMount: (container) => prepare(container.getState().scene),
     resetAfterCommit: () => {},
     shouldSetTextContent: () => false,
     clearContainer: () => false,
-    detachDeletedInstance: () => {},
-    hideInstance(instance: Instance) {
+    hideInstance(instance) {
       // Deatch while the instance is hidden
       const { attach: type, parent } = instance?.__r3f ?? {}
       if (type && parent) detach(parent, instance, type)
       if (instance.isObject3D) instance.visible = false
       invalidateInstance(instance)
     },
-    unhideInstance(instance: Instance, props: InstanceProps) {
+    unhideInstance(instance, props) {
       // Re-attach when the instance is unhidden
       const { attach: type, parent } = instance?.__r3f ?? {}
       if (type && parent) attach(parent, instance, type)
       if ((instance.isObject3D && props.visible == null) || props.visible) instance.visible = true
       invalidateInstance(instance)
     },
-    createTextInstance: () => {},
+    createTextInstance: () => {
+      throw new Error('Text is not allowed in the R3F tree.')
+    },
     hideTextInstance: () => {
       throw new Error('Text is not allowed in the R3F tree.')
     },
     unhideTextInstance: () => {},
-    getCurrentEventPriority: () => (getEventPriority ? getEventPriority() : DefaultEventPriority),
-    // @ts-ignore
```

Comment:
Did you mean to remove this? What about `supportsMicrotask`?

Reply:
Yeah, I didn't find these properties in Reconciler types. I'm not sure if they have been removed by the React team or if this is a bug in the types.

Reply:
I see. I was curious about the latter case since we didn't implement `scheduleMicrotask` (so `supportsMicrotask` indeed looked out of place), but the source of truth is unfortunately still the JS implementation. These would be the supported methods of 0.27: https://github.com/facebook/react/blob/34aa5cfe0d9b6ec4667e02bf46ab34d83dfb2d6d/packages/react-reconciler/src/forks/ReactFiberHostConfig.custom.js.

Quite a lot of these are for experimental (via feature flags), DOM/react-server, and testing APIs. With some digging, I only found `getCurrentEventPriority` to be pertinent to R3F, so we'll have to add that back. Everything else looks good to me.